### PR TITLE
update solid mechanics, scripts, cross_product, schemes, models..

### PR DESCRIPTION
**Pull request by josep-m-carbonell**

### CODE REVIEW

File: kratos/tests/geometries/test_quadrilateral_3d_4.cpp
Comment (roigcarlo) [inline kratos/tests/geometries/test_quadrilateral_3d_4.cpp]: Why has this been included?

File: applications/PFEMapplication/CMakeLists.txt~
Comment (roigcarlo) [inline applications/PFEMapplication/CMakeLists.txt~:46]: This cannot be deleted...

**Review by roigcarlo — DISMISSED** [review]

File: applications/PFEMapplication/CMakeLists.txt~
Comment (roigcarlo) [reply applications/PFEMapplication/CMakeLists.txt~:46]: Forget about this, i didn't saw the ~

File: kratos/tests/geometries/test_quadrilateral_3d_4.cpp
Comment (josep-m-carbonell) [reply kratos/tests/geometries/test_quadrilateral_3d_4.cpp]: I don't know about this change. I can not remember if I made it, but I guess I don't. It is possible that this line was there before and it was cleaned but in some merge remained there in my local branch. However it is strange. If it is not needed it must be taken away.

File: applications/PFEMapplication/CMakeLists.txt~
Comment (josep-m-carbonell) [reply applications/PFEMapplication/CMakeLists.txt~:46]: Yes this application was deleted some time ago.

File: kratos/tests/geometries/test_quadrilateral_3d_4.cpp
Comment (RiccardoRossi) [reply kratos/tests/geometries/test_quadrilateral_3d_4.cpp]: @josep-m-carbonell are u still using GitKraken? i am asking since we are reaching the conclusion that it is too easy to make mistakes when using it, since it brings to one click "strange" operations.

**Review by lluis-mv — DISMISSED** [review]
I directly made some modifications (modified cam clay and the script to write the json in PfemSolid problemtype).
In general it is OK with me.

File: applications/PFEMapplication/CMakeLists.txt~
Comment (josep-m-carbonell) [reply applications/PFEMapplication/CMakeLists.txt~:46]: @roigcarlo  it stills blocks the pr

Comment (RiccardoRossi) [reply applications/PFEMapplication/CMakeLists.txt~:46]: I am not involved in this review, so i will not say anything about the code.
However as a comment, it would be very good if you avoid this sort of enormous PRs.
A PR should include whenever possible only a few files or specific modifications. In this one you are merging a lot of different PRs into your own branch prior to merging it to the master. This makes the reviewing much more difficult and slower

**Review by lluis-mv — APPROVED** [review]

**Comment by pooyan-dadvand** [comment]
@josep-m-carbonell I would strongly suggest you to avoid such a huge PR and even worse before the beginning of the holidays.

**Pull request merged**
